src/common/hist_util.cc

```diff
@@ -1090,13 +1127,27 @@ void GHistBuilder::BuildBlockHist(const std::vector<GradientPair>& gpair,
       const GradientPair stat = gpair[rid];
       for (size_t j = ibegin; j < iend; ++j) {
         const uint32_t bin = gmat.index[j];
-        p_hist[bin].Add(stat);
+        p_hist[bin].Add(stat.GetGrad(), stat.GetHess());
```

Comment:
Why not get this into `GradientPair` with `operator+=`?

Reply:
no possibility to call  `GradientPairInternal<float>::operator+=(const  GradientPairInternal<double>&)`